benchmarks/benchmarks/loading.py

```diff
@@ -0,0 +1,186 @@
+                    x, y, z, 1, compress, file_type_args
+                )
+        return file_path_dict
+
+    def setup(
+        self,
+        file_path_dict: dict,
+        xyz: tuple,
+        compress: bool,
+        file_format: str,
+    ) -> None:
+        self.file_path = file_path_dict[xyz][compress][file_format]
+        self.cube = self.load()
+
+    def load(self) -> Cube:
+        return load_cube(self.file_path)
+
+    def time_load(self, _, __, ___, ____) -> None:
+        _ = self.load()
+
+    def time_realise(self, _, __, ___, ____) -> None:
+        # Cache the original data object and re-apply after realisation, which
+        #  restores original state for the next repeat. The cache/apply steps
+        #  add negligible time to the benchmark result.
+        data_original = self.cube.core_data()
+        _ = self.cube.data
+        self.cube.data = data_original
+
+
+class STASHConstraint:
+    # xyz sizes mimic LoadAndRealise to maximise file re-use.
+    params = [[(2, 2, 2), (1280, 960, 5)], ["FF", "PP"]]
+    param_names = ["xyz", "file_format"]
+
+    def setup_cache(self) -> dict:
+        file_type_args = self.params[1]
+        file_path_dict = {}
+        for xyz in self.params[0]:
+            x, y, z = xyz
+            file_path_dict[xyz] = create_um_files(
+                x, y, z, 1, False, file_type_args
+            )
+        return file_path_dict
+
+    def setup(
+        self, file_path_dict: dict, xyz: tuple, file_format: str
+    ) -> None:
+        self.file_path = file_path_dict[xyz][file_format]
+
+    def time_stash_constraint(self, _, __, ___) -> None:
+        _ = load_cube(self.file_path, AttributeConstraint(STASH="m??s??i901"))
+
+
+class TimeConstraint:
+    params = [[3, 20], ["FF", "PP", "NetCDF"]]
+    param_names = ["time_dim_len", "file_format"]
+
+    def setup_cache(self) -> dict:
+        file_type_args = self.params[1]
+        file_path_dict = {}
+        for time_dim_len in self.params[0]:
+            file_path_dict[time_dim_len] = create_um_files(
+                20, 20, 5, time_dim_len, False, file_type_args
+            )
+        return file_path_dict
+
+    def setup(
+        self, file_path_dict: dict, time_dim_len: int, file_format: str
+    ) -> None:
+        self.file_path = file_path_dict[time_dim_len][file_format]
+        self.time_constr = Constraint(time=lambda cell: cell.point.year < 3)
+
+    def time_time_constraint(self, _, __, ___) -> None:
+        _ = load_cube(self.file_path, self.time_constr)
+
+
+class ManyVars:
+    @staticmethod
+    def _create_file(save_path: str) -> None:
+        """Is run externally - everything must be self-contained."""
+        import numpy as np
+
+        from iris import save
+        from iris.coords import AuxCoord
+        from iris.cube import Cube
+
+        data_len = 8
+        data = np.arange(data_len)
+        cube = Cube(data, units="unknown")
+        extra_vars = 80
```

Comment:
Maybe, but here I'm mostly interested in migrating the coverage we already had offline, rather than improving coverage at the same time. Do you think it's important?

Reply:
> migrating the coverage we already had offline, rather than improving coverage at the same time

No, I think I have to agree there is no great need, and making it comparable to the existing has it's own value anyway.